policy-controller/k8s/index/src/index.rs

```diff
@@ -1070,13 +1082,57 @@ impl Pod {
             );
         };
 
+        // No Server is configured, so requests on the Pod's probe paths are
+        // authorized.
+        let http_routes = {
+            let probe_paths = probes.get(&port).into_iter().flatten().map(|p| p.as_str());
+            Self::http_probe_routes(probe_paths, &config.probe_networks)
+        };
+
         InboundServer {
             reference: ServerRef::Default(policy.to_string()),
             protocol,
             authorizations,
-            http_routes: HashMap::default(),
+            http_routes,
         }
     }
+
+    fn http_probe_routes<'p>(
+        paths: impl Iterator<Item = &'p str>,
+        networks: &[IpNet],
+    ) -> HashMap<RouteRef, InboundHttpRoute> {
+        if networks.is_empty() {
+            return HashMap::default();
+        }
+        paths
+            .map(|path| {
+                let authz = Some((
+                    AuthorizationRef::Default("probe".to_string()),
+                    ClientAuthorization {
+                        networks: networks.iter().copied().map(Into::into).collect(),
+                        authentication: ClientAuthentication::Unauthenticated,
+                    },
+                ));
+
+                let route = InboundHttpRoute {
+                    hostnames: vec![],
+                    rules: vec![InboundHttpRouteRule {
+                        matches: vec![HttpRouteMatch {
+                            path: Some(PathMatch::Exact(path.to_string())),
+                            headers: vec![],
+                            query_params: vec![],
+                            method: Some(Method::GET),
+                        }],
+                        filters: vec![],
+                    }],
+                    authorizations: authz.into_iter().collect(),
+                    creation_timestamp: std::time::SystemTime::now().into(),
```

Comment:
> hmm, is this right? technically, the pod spec is the kubernetes resource that defined the probe routes, so it might be theoretically more correct for the probe to inherit the pod spec's creation timestamp...

i think in reality it doesn't actually matter--we'll only use these routes when there are no other routes, so there's no possibility of needing to order things.  However, I do think it would be better to use `std::time::SystemTime::UNIX_EPOCH` so that it's a stable time value, for predictability.

> part of me wonders if we might want to move the timestamp out of the InboundHttpRoute type and into the RouteRef type, or something, and then do the ordering based on that instead?

I don't think that's right -- the routeref is part of a map key.

Reply:
actually, `UNIX_EPOCH` would mean that we always prefer them to resource routes, which isn't what we want. Perhaps best to make the creation_timestamp an `Option` (which means we can stop using fake times if they're not present in the spec).